doc/reference/trd-syscalls.md

```diff
@@ -624,6 +627,20 @@ The exit identifier specifies which call is invoked.
 | exit-terminate  |                     0 |
 | exit-restart    |                     1 |
 
+The difference between `exit-terminate` and `exit-restart` is what behavior
+the application asks from the kernel. With `exit-terminate`, the application
+tells the kernel that it considers itself completed and does not need to run
+again. With `exit-restart`, it tells the kernel that it would like to be
+rebooted and run again. For example, `exit-terminate` might be used by a 
+process that stores some one-time data on flash, while `exit-restart` might
+be used if the process runs out of memory.
+
+The completion code is an unsigned 32-bit number which indicates status. This
+information can be stored in the kernel and used in management or policy decisions.
+The definition of these status codes is outside the scope of this document.
+
+If an exit syscall is successful, it does not return. Therefore, the return
+value of an exit syscall is always _Failure_.
```

Comment:
If I'm reading the code correctly, `exit-terminate` and `exit-restart` never return regardless of whether they fail. Is that correct?

Do we want to guarantee that behavior for `exit-terminate`? It would make the entry point logic slightly simpler.

Reply:
`exit-terminate` and `exit-restart` always succeed and so should never return. We should guarantee it for both of them. I think going down the "conditional restart" path is probably not helpful (e.g., "well exit restart only if you'll restart me").

Reply:
I've updated the document to say that that `exit-restart` and `exit-terminate` MUST succeed (and so never return).